### PR TITLE
Update macOS for January 2024

### PR DESCRIPTION
**Pull request by BillyONeal**
macOS 14.2.1 on amd64, 14.3 on arm64
XCode Console Tools 15.1

Other changes:
* macOS install is done from the installers rather than from the recovery partition
* readme.md is now written for a vcpkg-team audience
* vagrant is removed as a dependency so that discussion of how to persist the VMs between machines could be included
* arm64 is added using https://github.com/s-u/macosvm

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -279,8 +277,6 @@ directxsdk:x64-windows=skip
 directxsdk:x64-windows-static-md=skip
 discord-rpc:arm64-uwp=fail
 discord-rpc:x64-uwp=fail
-discordcoreapi:x64-linux=fail
```
Comment (Cheney-W) [inline scripts/ci.baseline.txt:282]: This record still needs to be kept.

**Comment by Cheney-W** [comment]
discordcoreapi:x64-linux still failed with below error:
`/mnt/vcpkg-ci/buildtrees/discordcoreapi/src/v2.0.3-443fae93ea.clean/Library/../Include/discordcoreapi/Utilities/Base.hpp:409:46: error: ‘current_zone’ is not a member of ‘std::chrono’`

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (BillyONeal) [reply scripts/ci.baseline.txt:282]: Yeah I messed up a merge :/

**Comment by BillyONeal** [comment]
@autoantwort pointed out I created the VMs with the wrong number of cpus and RAM over Discord so I have severed the arm64 part into https://github.com/microsoft/vcpkg/pull/35665

**Comment by BillyONeal** [comment]
The amd64 instructions are toast due to popups happening during the build that cause things to hang while building `kf5texteditor`:

screenshot popup example

**Comment by autoantwort** [comment]
> -- installation will execute /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/x64-osx/debug/lib/libexec/kauth/kauth-policy-gen buffer/org.kde.ktexteditor.katetextbuffer.actions in /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/kf5texteditor/src/v5.98.0-94e36b3867.clean/src

I have done that manually, but it simply runs without a popup. Maybe I already said no in the past (?)

**Comment by BillyONeal** [comment]
OK, now everything happens through SSH just like it did with Vagrant.

**Comment by BillyONeal** [comment]
I made the arm64 boxes incorrectly (forgot to expand the disk) so if amd64 passes I'm going to just use that.

**Comment by autoantwort** [comment]
> I made the arm64 boxes incorrectly (forgot to expand the disk)

But apart from that, everything seems to be working now?

**Comment by BillyONeal** [comment]
> > I made the arm64 boxes incorrectly (forgot to expand the disk)
> 
> But apart from that, everything seems to be working now?

I don't know, since it got about 200 packages in and exploded due to no disk space

**Comment by BillyONeal** [comment]
REGRESSION: lensfun:x64-osx failed with BUILD_FAILED. If expected, add lensfun:x64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

This matches https://dev.azure.com/vcpkg/public/_build/results?buildId=98724 so I'm not looking at it.

REGRESSION: allegro5:arm64-osx failed with BUILD_FAILED. If expected, add allegro5:arm64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.
REGRESSION: allegro5:x64-osx failed with BUILD_FAILED. If expected, add allegro5:x64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
-- Performing Test TTF_COMPILES
CMake Error at /Users/vcpkg/Data/b/allegro5/arm64-osx-dbg/CMakeFiles/CMakeScratch/TryCompile-AkvfjL/CMakeLists.txt:25 (target_link_libraries):
  Target "cmTC_c6a8b" links to:

    ZLIB::ZLIB

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

This is because allegro5 is doing:

```cmake
if(SUPPORT_FONT AND WANT_TTF)
    find_package(Freetype)

    option(FREETYPE_ZLIB "Enable if FreeType static library needs zlib linked in." off)
    option(FREETYPE_BZIP2 "Enable if FreeType static library needs bzip2 linked in." off)
    option(FREETYPE_PNG "Enable if FreeType static library needs png linked in." off)
    option(FREETYPE_HARFBUZZ "Enable if FreeType static library needs harfbuzz linked in." off)
	
    if(FREETYPE_FOUND)
        set(FREETYPE_TEST_SOURCE "
            #include <ft2build.h>
            #include FT_FREETYPE_H
            int main(void) {
                FT_Library ft;
                FT_Open_Args args;
                FT_Face face;
                FT_Int32 ft_load_flags;
                FT_Error e;
                int ft_index;
                FT_Init_FreeType(&ft);
                FT_Open_Face(ft, &args, 0, &face);
                return FT_Load_Glyph(face, ft_index, ft_load_flags);
            }")

        set(TTF_INCLUDE_DIRECTORIES ${FREETYPE_INCLUDE_DIRS})

        set(CMAKE_REQUIRED_LIBRARIES ${FREETYPE_LIBRARIES})  # <----- !!!!!!!
        set(CMAKE_REQUIRED_INCLUDES ${FREETYPE_INCLUDE_DIRS})
        run_c_compile_test("${FREETYPE_TEST_SOURCE}" TTF_COMPILES)
```

and `FREETYPE_LIBRARIES` contains ZLIB::ZLIB. I'm not sure how to fix this problem exactly but it doesn't look related to the macos version change...

REGRESSION: libgo:x64-osx failed with BUILD_FAILED. If expected, add libgo:x64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
/Users/vcpkg/Data/b/libgo/src/c91da23096-01178f7311.clean/libgo/netio/unix/hook.cpp:652:5: error: declaration of 'select' has a different language linkage
int select(int nfds, fd_set *readfds, fd_set *writefds,
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.2.sdk/usr/include/sys/_select.h:43:10: note: previous declaration is here
int      select(int, fd_set * __restrict, fd_set * __restrict,
         ^
1 error generated.
```

REGRESSION: libgwenhywfar:arm64-osx failed with BUILD_FAILED. If expected, add libgwenhywfar:arm64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
duplicate symbol '_aes_v8_ctr32_encrypt_blocks' in:
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libgnutls.a[251](aes-aarch64.o)
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libcrypto.a[10](libcrypto-lib-aesv8-armx.o)
duplicate symbol '_aes_v8_encrypt' in:
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libgnutls.a[251](aes-aarch64.o)
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libcrypto.a[10](libcrypto-lib-aesv8-armx.o)
duplicate symbol '_aes_v8_cbc_encrypt' in:
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libgnutls.a[251](aes-aarch64.o)
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libcrypto.a[10](libcrypto-lib-aesv8-armx.o)
duplicate symbol '_aes_v8_decrypt' in:
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libgnutls.a[251](aes-aarch64.o)
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libcrypto.a[10](libcrypto-lib-aesv8-armx.o)
duplicate symbol '_aes_v8_set_encrypt_key' in:
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libgnutls.a[251](aes-aarch64.o)
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libcrypto.a[10](libcrypto-lib-aesv8-armx.o)
duplicate symbol '_aes_v8_set_decrypt_key' in:
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libgnutls.a[251](aes-aarch64.o)
    /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libcrypto.a[10](libcrypto-lib-aesv8-armx.o)
ld: 6 duplicate symbols
```

The library explicitly depends on both openssl and guntls so this seems like a problem in upstream, so I've marked it unsupported.

REGRESSION: libopensp:arm64-osx failed with BUILD_FAILED. If expected, add libopensp:arm64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.
REGRESSION: libopensp:x64-osx failed with BUILD_FAILED. If expected, add libopensp:x64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
In file included from .././../src/1.5.2-e9889c48a1.clean/lib/Dtd.cxx:8:
In file included from .././../src/1.5.2-e9889c48a1.clean/include/Dtd.h:11:
In file included from .././../src/1.5.2-e9889c48a1.clean/include/NamedResourceTable.h:11:
.././../src/1.5.2-e9889c48a1.clean/include/Ptr.h:72:11: error: ISO C++11 does not allow access declarations; use using declarations instead
  Ptr<T>::isNull;
          ^
  using 
.././../src/1.5.2-e9889c48a1.clean/include/Ptr.h:73:11: error: ISO C++11 does not allow access declarations; use using declarations instead
  Ptr<T>::clear;
          ^
  using 
```

Almost certainly caused by compiler version upgrade.

REGRESSION: nana:arm64-osx failed with BUILD_FAILED. If expected, add nana:arm64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
CMake Error at /opt/homebrew/Cellar/cmake/3.28.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find X11 (missing: X11_X11_INCLUDE_PATH X11_X11_LIB)
Call Stack (most recent call first):
```

Might be fixable with more homebrew libs but the x64 one is already skipped.

REGRESSION: oatpp-libressl:arm64-osx failed with BUILD_FAILED. If expected, add oatpp-libressl:arm64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
CMake Error at CMakeLists.txt:104 (message):
  OpenSSL installed instead of LibreSSL.  oatpp-libressl requires LibreSSL.
```

This is already skipped on x64 by virtue of libressl being skipped.

REGRESSION: openturns:arm64-osx failed with BUILD_FAILED. If expected, add openturns:arm64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
CMake Error at /opt/homebrew/Cellar/cmake/3.28.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find BISON: Found unsuitable version "2.3", but required is at
  least "2.4" (found /usr/bin/bison)
```

This was also skipped on x64. It's using `/usr/bin/bison` rather than `/opt/homebrew/opt/bison/bin/bison` because the `vcpkg_add_to_path` calls are appending rather than prepending.

REGRESSION: optional-lite:arm64-osx failed with FILE_CONFLICTS. If expected, add optional-lite:arm64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
error: The following files are already installed in /Users/vcpkg/Data/installed/arm64-osx and are in conflict with optional-lite:arm64-osx
Installed by optional-bare:arm64-osx  
include/nonstd/optional.hpp
```

Already skipped on all other platforms.

REGRESSION: zydis:x64-osx failed with BUILD_FAILED. If expected, add zydis:x64-osx=fail to /Users/vcpkg/Data/work/1/s/scripts/azure-pipelines/../ci.baseline.txt.

```console
ld: pointer not 4-byte aligned at __DATA_CONST+0x79, fix alignment or disable chained fixups
final section layout:
    __PAGEZERO           addr=0x00000000, size=0x100000000, fileOffset=0x00000000
    __TEXT               addr=0x100000000, size=0x00084000, fileOffset=0x00000000
        __text           addr=0x1000045c0, size=0x0001951f, fileOffset=0x000045c0
        __stubs          addr=0x10001dadf, size=0x0000003c, fileOffset=0x0001dadf
        __cstring        addr=0x10001db1b, size=0x0000522a, fileOffset=0x0001db1b
        __const          addr=0x100022d50, size=0x00061165, fileOffset=0x00022d50
        __unwind_info    addr=0x100083eb8, size=0x00000140, fileOffset=0x00083eb8
    __DATA_CONST         addr=0x100084000, size=0x00008000, fileOffset=0x00084000
        __got            addr=0x100084000, size=0x00000068, fileOffset=0x00084000
        __const          addr=0x100084070, size=0x000050ed, fileOffset=0x00084070
    __LINKEDIT           addr=0x10008c000, size=0x00008000, fileOffset=0x0008c000
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Looks like https://github.com/zyantific/zydis/commit/a605f54580c26d2164530fab323057834569d169 :)

**Comment by BillyONeal** [comment]
@autoantwort The only one I'm not sure how to address is the `allegro5` one. Any ideas?

**Comment by autoantwort** [comment]
> @autoantwort The only one I'm not sure how to address is the `allegro5` one. Any ideas?

Hm no. I also can not reproduce

**Comment by BillyONeal** [comment]
> > @autoantwort The only one I'm not sure how to address is the `allegro5` one. Any ideas?
> 
> Hm no. I also can not reproduce

image

Maybe depends on CMake version?

**Comment by BillyONeal** [comment]
> > > @autoantwort The only one I'm not sure how to address is the `allegro5` one. Any ideas?
> > 
> > 
> > Hm no. I also can not reproduce
> 
> image
> 
> Maybe depends on CMake version?

Confirmed: It is broken in CMake 3.28 but works in 3.27.

**Comment by BillyONeal** [comment]
In CMake 3.27, `FREETYPE_LIBRARIES` is:

optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libfreetype.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libfreetyped.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libz.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libz.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbz2.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbz2d.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libpng16.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libpng16d.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libz.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libz.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbrotlidec.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbrotlidec.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbrotlicommon.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbrotlicommon.a

while in 3.28 it is:

optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libfreetype.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libfreetyped.a;$<LINK_ONLY:ZLIB::ZLIB>;$<$<NOT:$<CONFIG:DEBUG>>:/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbz2.a>;$<$<CONFIG:DEBUG>:/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbz2d.a>;$<$<NOT:$<CONFIG:DEBUG>>:/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libpng16.a>;$<$<CONFIG:DEBUG>:/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libpng16d.a>;$<$<NOT:$<CONFIG:DEBUG>>:/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libz.a>;$<$<CONFIG:DEBUG>:/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libz.a>;$<$<NOT:$<CONFIG:DEBUG>>:/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbrotlidec.a>;$<$<CONFIG:DEBUG>:/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbrotlidec.a>;$<$<NOT:$<CONFIG:DEBUG>>:/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbrotlicommon.a>;$<$<CONFIG:DEBUG>:/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbrotlicommon.a>;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libz.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libz.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbz2.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbz2d.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libpng16.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libpng16d.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libz.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libz.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbrotlidec.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbrotlidec.a;optimized;/Users/vcpkg/vcpkg/installed/arm64-osx/lib/libbrotlicommon.a;debug;/Users/vcpkg/vcpkg/installed/arm64-osx/debug/lib/libbrotlicommon.a

The only place I see this `LINK_ONLY` is in `installed/arm64-osx/share/freetype/freetype-targets.cmake`:

```cmake
set_target_properties(freetype PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:ZLIB::ZLIB>;\$<\$<NOT:\$<CONFIG:DEBUG>>:${VCPKG_IMPORT_PREFIX}/lib/libbz2.a>;\$<\$<CONFIG:DEBUG>:${VCPKG_IMPORT_PREFIX}/debug/lib/libbz2d.a>;\$<\$<NOT:\$<CONFIG:DEBUG>>:${VCPKG_IMPORT_PREFIX}/lib/libpng16.a>;\$<\$<CONFIG:DEBUG>:${VCPKG_IMPORT_PREFIX}/debug/lib/libpng16d.a>;\$<\$<NOT:\$<CONFIG:DEBUG>>:${VCPKG_IMPORT_PREFIX}/lib/libz.a>;\$<\$<CONFIG:DEBUG>:${VCPKG_IMPORT_PREFIX}/debug/lib/libz.a>;\$<\$<NOT:\$<CONFIG:DEBUG>>:${VCPKG_IMPORT_PREFIX}/lib/libbrotlidec.a>;\$<\$<CONFIG:DEBUG>:${VCPKG_IMPORT_PREFIX}/debug/lib/libbrotlidec.a>;\$<\$<NOT:\$<CONFIG:DEBUG>>:${VCPKG_IMPORT_PREFIX}/lib/libbrotlicommon.a>;\$<\$<CONFIG:DEBUG>:${VCPKG_IMPORT_PREFIX}/debug/lib/libbrotlicommon.a>"
)
```

**Comment by BillyONeal** [comment]
I think https://gitlab.kitware.com/cmake/cmake/-/commit/d83d925045484c882527477b01596073f45d3faa is the source of the regression

**Comment by BillyONeal** [comment]
I filed https://gitlab.kitware.com/cmake/cmake/-/issues/25635

**Comment by autoantwort** [comment]
I think we should also move the triplet file from the community triplet folder to the official one.

**Comment by BillyONeal** [comment]
> I think we should also move the triplet file from the community triplet folder to the official one.

I agree but I need lots of folks to sign off making that change so I want to land this as is for now.

**Review by JavierMatosD — APPROVED** [review]

**Comment by autoantwort** [comment]
Whoop whoop 🎉 Thank you for investing this large amount of time to make this possible! :)

**Comment by BillyONeal** [comment]
> Whoop whoop 🎉 Thank you for investing this large amount of time to make this possible! :)

Thank *you* for the large amount of time fixing a bunch of ports before we got off our butts to fix it so that doing it was relatively painless :)

**Pull request merged**